3529/2/Lib/checkout_context.rb

```diff
@@ -0,0 +1,19 @@
+require 'fileutils'
+require './Lib/Utiles/path_loader.rb'
+require './Lib/Utiles/coordinates_handler.rb'
+require './Lib/Utiles/utiles.rb'
+
+module CheckoutContext
+  include ImageFileWriter
+  include CoordinatesHandler
+
+  def checkout!(message = nil, *)
+    @path_check = PathLoader.new(payload['from']['id'], payload['chat']['id'])
+    checko(StatementFactory.build(message))
+  end
+
+  def checko(message)
+    public_send("reply_to_#{message.class}".downcase.to_sym, @path_check, 'out')
```

Comment:
OOOOOO!!!!!!!
Explain it to me, please))))

Reply:
I found this article: http://designisrefactoring.com/2015/03/09/refactoring-away-a-conditional/
To make long story short that will choose dynamic the right method for each case, depending on wich class was created. i can call method by its name with the help of public_send method.